ckanext/data_qld/tests/conftest.py

```diff
@@ -186,3 +186,73 @@ def do_not_validate(monkeypatch):
     monkeypatch.setattr(
         "ckanext.validation.utils.is_resource_could_be_validated",
         lambda a, b: False)
+
+
+class Comment(factory.Factory):
+    """A factory class for creating ytp comment. It must accept user_id and
+    package_name, because I don't want to create extra entities in database
+    during tests"""
+
+    class Meta:
+        model = ytp_model.Comment
+
+    user_id = None
+    entity_type = "dataset"
+    entity_name = None
+
+    subject = "comment-subject"
+    comment = "comment-text"
+
+    @classmethod
+    def _build(cls, target_class, *args, **kwargs):
+        raise NotImplementedError(".build() isn't supported in CKAN")
+
+    @classmethod
+    def _create(cls, target_class, *args, **kwargs):
+        if args:
+            assert False, "Positional args aren't supported, use keyword args."
+
+        kwargs["url"] = "/{}/{}".format(kwargs["entity_type"], kwargs["entity_name"])
+
+        return helpers.call_action(
+            "comment_create", context={"user": kwargs["user_id"], "ignore_auth": True}, **kwargs
+        )
+
+
+@pytest.fixture
+def comment_factory():
```

Comment:
Why does this fixture exist when no test is using it?

Reply:
this may be from the ytp comments which was moved.